### ISSUE TITLE
Support custom trade parameters for exchanges

### ISSUE THREAD

**Issue by knowncolor** [knowm/XChange#3889]
This is a very exciting project, I'm just getting started using it.

I am interested in using a parameter, TimeToLive (TTL) that certain exchanges support for limit orders. How should I go about implementing this feature, or is it possible to add custom parameters to orders already?

Thanks!

**Comment by earce** [comment]
is TTL part of the REST api, like a query or path param?

**Comment by knowncolor** [comment]
> is TTL part of the REST api, like a query or path param?

Yes it is, see https://github.com/coinfloor/API/blob/master/BIST.md#buy-limit-order for example

**Comment by earce** [comment]
I would imagine you can just modify [CoinfloorAuthenticated.java](https://github.com/knowm/XChange/blob/develop/xchange-coinfloor/src/main/java/org/knowm/xchange/coinfloor/CoinfloorAuthenticated.java) to include the param you want.

**Comment by walec51** [comment]
cast a `*Service` class to its `*ServieRaw` equivalent to gain access to exchange specific features - if they are missing implement them in the  `*ServieRaw` classes

generic interfaces / dtos should have features that are supported by most exchanges - to add something there your should first do an analysis there are they supported: https://github.com/knowm/XChange/wiki/Generic-feature-candidates

**Comment by knowncolor** [comment]
Makes total sense, I'll open a request to add the missing `ttl` parameter.

**Comment by nielsdraaisma** [comment]
Most exchanges do this using the IOrderFlags interface, depending on the exchange the flags can be used to trigger IOC and MOC order types.
